### PR TITLE
Allow `.` and `~` chars in registration tokens

### PR DESCRIPTION
**Pull request by govynnus**
See https://github.com/matrix-org/matrix-doc/pull/3231/commits/79939f955b34113404b5d363a981f790aa7ec0e7

Signed-off-by: Callum Brown <callum@calcuode.com>

### CODE REVIEW

**Comment by anoadragon453** [comment]
Looks like the unit test `test_create_unable_to_generate_token` is now failing. @govynnus could you take a look?

**Comment by govynnus** [comment]
Hmm, I did run the unit tests before pushing, but I'll take a look

**Comment by govynnus** [comment]
Oh it must be because it now doesn't add all possible 1 char tokens to the db, so there's a possibility creating one succeeds.

**Comment by anoadragon453** [comment]
@govynnus Oh, right indeed! Lucky we caught it here then :)

File: changelog.d/10887.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Allow `.` and `~` characters when creating registration tokens as per the [change to MSC3231](https://github.com/matrix-org/matrix-doc/pull/3231/commits/79939f955b34113404b5d363a981f790aa7ec0e7).
```
Comment (clokep) [inline changelog.d/10887.bugfix:1]: I think we usually just link to the MSC, not the specific commit in it?

**Review by clokep — APPROVED** [review]

File: changelog.d/10887.bugfix (hunk 1 above)
Comment (clokep) [reply changelog.d/10887.bugfix:1]:
```suggestion
Allow the `.` and `~` characters when creating registration tokens as per the change to [MSC3231](https://github.com/matrix-org/matrix-doc/pull/3231).
```

Comment (govynnus) [reply changelog.d/10887.bugfix]: Ok, thanks

**Pull request merged**